### PR TITLE
PROB-1137 - Harmony Connection timed out

### PR DESCRIPTION
**Pull request by juano2310**

### CODE REVIEW

**Comment by ron2911** [comment]
The code looks good to me. I recommend another set of eyes, because I don't have much context on Harmony stuff.

@marstorp Can you please review this too?

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 1]
```diff
@@ -349,7 +349,7 @@ def initialize() {
 	if (selectedhubs || selectedactivities) {
 		addDevice()
     runEvery5Minutes("poll")
-    getActivityList()
+    log.trace getActivityList()
```
Comment (marstorp) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: indentation spaces vs. tabs

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 2]
```diff
@@ -383,30 +383,35 @@ Map discoverDevices() {
...
-	    }
-	} catch (groovyx.net.http.HttpResponseException e) {
-        if (e.statusCode == 401) { // token is expired
-            state.remove("HarmonyAccessToken")
-            log.warn "Harmony - Harmony Access token has expired"
-        }
-	} catch (java.net.SocketTimeoutException e) {
+    def tokenParam = [auth: state.HarmonyAccessToken]
+    def url = "https://home.myharmony.com/cloudapi/activity/all?${toQueryString(tokenParam)}"
+	def params = [
```
Comment (ron2911) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: indentation is funky here.

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 3]
```diff
@@ -383,30 +383,35 @@ Map discoverDevices() {
...
+	} else {
 		log.warn "Harmony - Connection to the hub timed out. Please restart the hub and try again."
-    state.resethub = true
- 	} catch (e) {
-    log.info "Harmony - Error: $e"
+		state.resethub = true
+	}
+  } else {
+		if (response.status == 200) {
+	    	log.debug "Harmony - valid Token"
```
Comment (ron2911) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: indentation.

File: smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy [hunk 4]
```diff
@@ -564,34 +569,42 @@ def pollResponse(response, data) {
...
-        } catch (groovyx.net.http.HttpResponseException e) {
-        	log.trace e
-        } catch (java.net.SocketTimeoutException e) {
-        	log.trace e
-		    } catch(Exception e) {
-        	log.trace e
-		    }
-    }
+		def tokenParam = [auth: state.HarmonyAccessToken]
+	    def url = "https://home.myharmony.com/cloudapi/activity/all?${toQueryString(tokenParam)}"
```
Comment (ron2911) [inline smartapps/smartthings/logitech-harmony-connect.src/logitech-harmony-connect.groovy]: indentation.

**Comment by marstorp** [comment]
There's still mixed tabs and spaces but otherwise LGTM ⛵️

**Pull request merged**